internal/services/groups/groups_data_source_test.go

```diff
@@ -70,6 +70,34 @@ func TestAccGroupsDataSource_returnAll(t *testing.T) {
 	})
 }
 
+func TestAccGroupsDataSource_securityEnabled(t *testing.T) {
+	data := acceptance.BuildTestData(t, "data.azuread_groups", "test")
+
+	data.DataSourceTest(t, []resource.TestStep{
+		{
+			Config: GroupsDataSource{}.securityEnabled(),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).Key("display_names.#").Exists(),
+				check.That(data.ResourceName).Key("object_ids.#").Exists(),
+			),
+		},
+	})
+}
+
+func TestAccGroupsDataSource_mailEnabled(t *testing.T) {
+	data := acceptance.BuildTestData(t, "data.azuread_groups", "test")
+
+	data.DataSourceTest(t, []resource.TestStep{
+		{
+			Config: GroupsDataSource{}.mailEnabled(),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).Key("display_names.#").Exists(),
+				check.That(data.ResourceName).Key("object_ids.#").Exists(),
+			),
+		},
+	})
+}
```

Comment:
These are good, can we also add a test for security_enabled + mail_enabled to make sure these work together as expected?